### PR TITLE
[DX-906]prevent 404 in version by creating version selector dynamically

### PR DESCRIPTION
**Pull request by yurisasuke**
[DX-906]

[DX-906]: https://tyktech.atlassian.net/browse/DX-906?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/page_available_since.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    replace_3lts = 'https://tyk.io/docs/3-lts'
+    replace_latest = 'https://tyk.io/docs'
+    modified_url = re.sub(version_pattern, '', url)
+    modified_url = modified_url.replace(replace_nightly, '')
+    modified_url = modified_url.replace(replace_3lts, '')
+    modified_url = modified_url.replace(replace_latest, '')
+    return modified_url
+
+
+# print(get_urls("latest_sitemap.xml"))
```
Comment (dcs3spp) [inline scripts/page_available_since.py]: do we need to keep the comment?

File: scripts/page_available_since.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+
+def get_site_maps():
+    available_since = {}
+    for version in versions:
+        url = "https://tyk.io/{version}/sitemap.xml".format(version=version["path"])
+        print("Getting sitemap for {url}".format(url=url))
+        response = requests.get(url, headers={'user-agent': 'insomnia/2023.4.0'})
+        if response.status_code != 200:
+            raise Exception("unable to fetch the sitemap")
```
Comment (dcs3spp) [inline scripts/page_available_since.py]: does `raise_for_status` do this?

File: scripts/page_available_since.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    parsed = BeautifulSoup(content, "lxml")
+    urls_from_xml = []
+    loc_tags = parsed.find_all('loc')
+    for loc in loc_tags:
+        urls_from_xml.append(replace_base_url(loc.get_text()))
+    urls_from_xml.sort()
+    return sorted(urls_from_xml)
+
+
+def replace_base_url(url: str):
```
Comment (dcs3spp) [inline scripts/page_available_since.py]:
```suggestion
def replace_base_url(url: str) -> str:
```

File: scripts/page_available_since.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+        for url in urls:
+            if url in available_since:
+                available_since[url].append(version["path"])
+            else:
+                available_since[url] = [version["path"]]
+        print("finished adding to dict urls for version {version}".format(version=version["name"]))
+    return available_since
+
+
+def get_urls(content: str):
```
Comment (dcs3spp) [inline scripts/page_available_since.py]:
```suggestion
def get_urls(content: str) -> list:
```

File: scripts/page_available_since.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    },
+    {"path": "/docs/nightly/",
+     "name": "Nightly",
+     }
+]
+
+filePath = "../tyk-docs/data/page_available_since.json"
+
+
+def process_and_write_to_file():
```
Comment (dcs3spp) [inline scripts/page_available_since.py]:
```suggestion
def process_and_write_to_file() -> None:
```

File: scripts/page_available_since.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+    available = get_site_maps()
+    with open(filePath, 'w') as file:
+        json.dump(available, file, indent=4)
+
+
+def get_site_maps():
```
Comment (dcs3spp) [inline scripts/page_available_since.py]:
```suggestion
def get_site_maps() -> dict:
```

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Hi @Keithwachira PR LTGM. I have made some minor suggestions

**Comment by yurisasuke** [comment]
replaced by https://github.com/TykTechnologies/tyk-docs/pull/3989

**Pull request closed without merging**